src/js/common/schemaform/fields/FileField.jsx

```diff
@@ -147,6 +160,22 @@ export default class FileField extends React.Component {
                         disabled={this.props.disabled}
                         readonly={this.props.readonly}/>
                     </div>}
+                  {!hasErrors && uiOptions.allowRename &&
+                  <div className="schemaform-file-attachment">
+                    <SchemaField
+                      name="attachmentName"
+                      required
+                      schema={itemSchema.properties.name}
+                      uiSchema={uiOptions.attachmentName}
+                      errorSchema={attachmentNameErrors}
+                      idSchema={attachmentNameSchema}
+                      formData={formData[index].name}
+                      onChange={(value) => this.onAttachmentNameChange(index, value)}
```

Comment:
Can we add a unit test to cover this name update functionality?

Reply:
Good idea, will add.